### ISSUE TITLE
(x.45.3) Drill Through and Click Behavior broken for SQL based Models

### ISSUE THREAD

**Issue by Tony-metabase** [metabase/metabase#29517]
**Describe the bug**
When you create a model and use it as a data source for questions you lose Drill Through and Click Behaviour

**Logs**
There are no logs in the logs but error pops in the Network Developer Tools -> Console

```
drilldown.js:412 Uncaught TypeError: Cannot read properties of undefined (reading 'fields')
    at drilldown.js:412:58
    at Y (drilldown.js:431:16)
    at question (zoom-drill.js:9:21)
    at we (ZoomDrill.ts:9:8)
    at Mode.ts:33:50
    at Array.flatMap (<anonymous>)
    at e.value (Mode.ts:33:33)
    at m.value (Visualization.jsx:238:14)
    at Visualization.jsx:277:12
    at SVGRectElement.t (apply_tooltips.js:513:11)
```

**To Reproduce**
Steps to reproduce the behavior (if you can reproduce the bug using the Sample Database, we will find the issue faster):
1. Go to New SQL Question -> Paste the below -> Save -> Turn into a Model
``` 
Select Orders."ID" AS "ID", 
       Orders."CREATED_AT" AS "CREATED_AT"
From Orders
```
2. Go to model Metadata -> Map ID to Order.ID -> Created_At to Orders.Created_AT -> You will end up with 100% metadata

<img width="1510" alt="image" src="https://user-images.githubusercontent.com/110378427/227523596-8387fcbd-4495-4b4d-b439-b593f7b95acb.png">

 # Broken Drill Through

1. Go to New Question -> Select Your Model AS Data Source -> Count and Group by Date -> Visualize as Line Chart:
2. Try to click on the chart and you will see there is no Drill Through and an error is popping in the console

<img width="1505" alt="image" src="https://user-images.githubusercontent.com/110378427/227524334-5e68e21d-7d4c-4dbd-8b86-e161f1763363.png">

 # Broken Click Behavior 

1. Go to new Dashboard -> Add the above Chart to the dashboard
2. Edit Dashboard -> Go to Chart Click Behavior -> Link to Dashboard -> Link to some other dashboard or the same dashboard you are in -> Save

<img width="1493" alt="image" src="https://user-images.githubusercontent.com/110378427/227524892-c56cdc60-fb43-4d64-95c0-9a67f2757eef.png">

3. Try to click on the chart so it takes you to the linked dashboard but you end up with the same error:
<img width="1508" alt="image" src="https://user-images.githubusercontent.com/110378427/227525344-c4b5dbb2-cfe7-4021-ada9-54ace36577b4.png">

**Expected behavior**
Drill Through and Click Behavior works

**Information about your Metabase Installation:**

Able to replicate on 1.45.3 and `master`

**Severity**
Brutal

**Additional context**
Could have opened this in 2 separate Bugs but i believe the same underlying problem is causing issues in Drill Through and Click Behavior

**Comment by ekroymann** [comment]
Is this a recently introduced bug?. Are there any versions of metabase, that are unaffected from this? Or any other workaround?

**Comment by nemanjaglumac** [comment]
Tested this with the previous versions.

Works on `x.45.2` ✅ 
Broken in `x.45.3` ❌

**Comment by nemanjaglumac** [comment]
Tested a bit further and the breaking change was introduced here

https://github.com/metabase/metabase/pull/28022/files#diff-fabafeb7f6b516bccea0a558b2e4163293b3c99a2606f8b3768eba1769bf55adL452

as part of #28022 cc @ranquild

**Comment by nemanjaglumac** [comment]
Breaking change has been reverted and will not be present in the x.46.x release

However, we have to leave this issue open because the bug is still present in `x.45.3`. I've updated the title accordingly.

**Comment by nemanjaglumac** [comment]
This is now fixed in [`v0.45.3.1`](https://github.com/metabase/metabase/releases/tag/v0.45.3.1) and [`v1.45.3.1`](https://github.com/metabase/metabase/releases/tag/v1.45.3.1)
